src/Reflection/ExtendedMethodReflection.php

```diff
@@ -22,4 +22,6 @@ public function getAsserts(): Assertions;
 
 	public function getSelfOutType(): ?Type;
 
+	public function returnsByReference(): ?bool;
```

Comment:
In order to be useful at all, ExtendedMethodReflection has to be marked with `@api`, BUT:

1) There has to be an exception in ApiClassImplementsRule so that this interface cannot be implemented.
2) The exception list in ApiClassImplementsRule has to be synced with ApiInterfaceExtendsRule - preferably DRY via a constant for example.
3) All these interfaces need to be mentioned here: https://github.com/ondrejmirtes/BackwardCompatibilityCheck/blob/7d5993a5de91ead729e52680b2722eee8549f49f/bin/roave-backward-compatibility-check.php#L237
4) Assertions also need to be marked with `@api`.

All of these point should be done in a separate PR first, then we can get back to this one. Thanks.